src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -1,13 +1,94 @@
 import React from 'react';
-import {getFollowing} from '../../../service/profile';
+import {Context} from '../../../context';
+import DiscoveryAction from '../../../context/actions/discoveryAction';
+import following from '../../../context/actions/following';
 import {getFollowedDomain} from '../../../service/domain';
+import {getFollowing} from '../../../service/profile';
 import {getFollowingTopic} from '../../../service/topics';
-import following from '../../../context/actions/following';
-import {Context} from '../../../context';
 
 const useDiscovery = () => {
   const [, followingDispatch] = React.useContext(Context).following;
   const [refreshing, setRefreshing] = React.useState(false);
+  const [discovery, discoveryDispatch] = React.useContext(Context).discovery;
+
+  const users = React.useMemo(() => {
+    return discovery.initialUsers.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [discovery]);
+
+  const exhangeFollower = (newUserLists, willFollow, userId) => {
+    const indexUser = newUserLists.findIndex((item) =>
+      item.user ? item.user.user_id === userId : item.user_id === userId
+    );
+    if (newUserLists[indexUser].user) {
+      newUserLists[indexUser].user.following = !!willFollow;
+      newUserLists[indexUser].user.is_followed = !!willFollow;
+    } else {
+      newUserLists[indexUser].following = !!willFollow;
+      newUserLists[indexUser].is_followed = !!willFollow;
+    }
+    return newUserLists[indexUser];
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `exhangeFollower` function modifies the `newUserLists` directly, which is not a good practice as it mutates the state directly. Instead, you should create a copy of the array and modify that. 

```diff
-    const indexUser = newUserLists.findIndex((item) =>
+    let updatedUserLists = [...newUserLists];
+    const indexUser = updatedUserLists.findIndex((item) =>
      item.user ? item.user.user_id === userId : item.user_id === userId
    );
    if (updatedUserLists[indexUser].user) {
      updatedUserLists[indexUser].user.following = !!willFollow;
      updatedUserLists[indexUser].user.is_followed = !!willFollow;
    } else {
      updatedUserLists[indexUser].following = !!willFollow;
      updatedUserLists[indexUser].is_followed = !!willFollow;
    }
-    return newUserLists[indexUser];
+    return updatedUserLists[indexUser];
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->